packages/bundler-plugin-core/src/utils/normalizePath.ts

```diff
@@ -34,8 +29,12 @@ export const normalizePath = (path: string, format: string): string => {
     )})`;
 
     // grab the ending delimiter and create a regex group for it
-    let endingDelimiter =
-      format.at(match.hashIndex + match.hashString.length) ?? "";
+    let endingDelimiter = "";
```

Comment:
Also correct, but not worth the performance win because of the size of string, array lookups are cheap because the n is small for this method in general.

Only worry is the extra allocations, but considering this should be immediately GC'd for the most part not a problem.

In general the optimization would be to do this in single pass, and explicitly iterate over `format` and collect the delimiters accordingly.